sklearn/isotonic.py

```diff
@@ -271,6 +272,7 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
         check_consistent_length(X, y, sample_weight)
         X, y = [check_array(x, ensure_2d=False) for x in [X, y]]
 
+        X = as_float_array(X)
```

Comment:
I removed the redundant call to check_array on X and am now just calling it on y. Still, I think the behavior on int inputs should be consistent if possible over the codebase.

@Henley13 have you already discussed this?